### PR TITLE
Allow podman connections in synchronize module

### PR DESCRIPTION
**Pull request by sshnaidm**
##### ISSUE TYPE

- Feature Pull Request
For fixing issues with molecule: https://github.com/ansible-community/molecule-podman/issues/45

##### COMPONENT NAME

synchronize
##### ADDITIONAL INFORMATION

Add `podman` connections as it's done with buildah and docker.

```paste below

```

### CODE REVIEW

File: plugins/action/synchronize.py [hunk 1]
```diff
@@ -66,7 +66,15 @@ def _format_rsync_rsh_target(self, host, path, user):
             return path
 
         # If using docker or buildah, do not add user information
-        if self._remote_transport not in ['docker', 'community.general.docker', 'community.docker.docker', 'buildah', 'containers.podman.buildah'] and user:
+        if self._remote_transport not in [
```
Comment (Akasurde) [inline plugins/action/synchronize.py:69]: Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

**Comment by Akasurde** [comment]
@sshnaidm Thanks for the contribution.

File: plugins/action/synchronize.py (hunk 1 above)
Comment (sshnaidm) [reply plugins/action/synchronize.py:69]: done, @Akasurde , thanks for review

Comment (Akasurde) [reply plugins/action/synchronize.py:69]: Thanks.

File: changelogs/fragments/229_add_podman_connection_plugin_to_synchronize.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - synchronize - add ``community.podman.podman`` to the list of supported
```
Comment (Akasurde) [inline changelogs/fragments/229_add_podman_connection_plugin_to_synchronize.yml:3]:
```suggestion
  - synchronize - add ``community.podman.podman`` to the list of supported connection plugins (https://github.com/ansible-community/molecule-podman/issues/45).
```

**Review by Akasurde — APPROVED** [review]
LGTM. @saito-hideki @quidame @justjais  Could you please review this? thanks in advance.

**Review by justjais — APPROVED** [review]

**Comment by quidame** [comment]
LGTM too. @Crystallic , do you have time and possibility to run your failing tests with this version ?

**Review by saito-hideki — APPROVED** [review]
Thank you for the PR! It looks good to me.

**Comment by Akasurde** [comment]
@sshnaidm Thanks for the contribution @quidame @saito-hideki @justjais Thanks for the review.

**Comment by tadeboro** [comment]
I did not check, but I have a feeling that this PR did not do enough. The original error stems from the https://github.com/sshnaidm/ansible.posix/blob/a03e652437434fd9f600071bf99332ef4dd04292/plugins/action/synchronize.py#L198-L206 and this PR left that part of the code intact.

**Pull request merged**